src/main/NWN/API/Object/NwObject.cs

```diff
@@ -224,7 +224,7 @@ public override string ToString()
 
     public bool Equals(NwObject other)
     {
-      if (ReferenceEquals(null, other))
+      if (other is null)
```

Comment:
I wasn't able to confirm if `other is null` is the same as `other != null`, or `ReferenceEquals(null, other)`.

This will break `null == OBJECT_INVALID`.

Reply:
Not sure how you came to that conclusion.  As for the changes to the new standards not sure how you came to that conclusion either.
https://devblogs.microsoft.com/dotnet/welcome-to-c-9-0/
https://www.c-sharpcorner.com/article/null-value-and-null-reference-handling-c-sharp6-to-c-sharp-9-new-features-day-1/

Reply:
What does the last option equate to?
```cs
      NwObject obj = new NwObject(INVALID);
      obj == null; // true
      ReferenceEquals(null, obj); // false
      obj is null // ??
```

Reply:
According to the docs in the URL in my last comment
```
obj == null
is 
obj is null
```
https://stackoverflow.com/questions/6417902/checking-if-an-object-is-null-in-c-sharp
```
in C# > 7.0 use

if (obj is null) ...

This will ignore any == or != defined by the object (unless of course you want to use them ...)

For not null use if (obj is object) and from C# 9 you can also use if (obj is not null)
```
Someone did say this
```
Simplest way is to do object.ReferenceEquals(null, data)

Since (null==data) is NOT guaranteed to work:
```
But was said to be untrue an redacted. Someone further down posted
```
As of C# 9 you can do

if (obj is null) { ... }
For not null use

if (obj is not null) { ... }
If you need to override this behaviour use == and != accordingly.
```
Or you can do this as of 8.0 but I didn't do this because it didn't look as readable to me as what you can do in 9.0.
```
As of C# 8 you can use the 'empty' property pattern (with pattern matching) to ensure an object is not null:

if (obj is { })
{
    // 'obj' is not null here
}
```
Someone beneath that linked to this URL to explain this
https://www.thomasclaudiushuber.com/2020/03/12/c-different-ways-to-check-for-null/

Reply:
Yep, I did just test this locally and came to the same conclusion (also generates the same IL code)

IMO considering the context, I'd rather leave these as `ReferenceEquals` just to be super explicit, given they are part of equality members.